examples/vision/deeplabv3_plus.py

```diff
@@ -0,0 +1,290 @@
+
+- [DeepLab: Semantic Image Segmentation with Deep Convolutional Nets, Atrous Convolution, and Fully Connected CRFs](https://arxiv.org/abs/1606.00915)
+"""
+
+"""
+## Download Dataset
+"""
+
+import os
+import cv2
+import numpy as np
+from glob import glob
+from scipy.io import loadmat
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+"""shell
+gdown https://drive.google.com/uc?id=1B9A9UCJYMwTL4oBEo4RZfbMZMaZhKJaz
+unzip -q instance-level-human-parsing.zip
+"""
+
+"""
+## Building Tensorflow Dataset
+
+For this example, we would be using 200 images from the instance-level human parsing
+dataset to train our model.
+"""
+
+IMAGE_SIZE = 512
+BATCH_SIZE = 4
+NUM_CLASSES = 20
+DATA_DIR = "./instance-level_human_parsing/instance-level_human_parsing/Training"
+MAX_IMAGES = 200
+
+train_images = sorted(glob(os.path.join(DATA_DIR, "Images/*")))[:MAX_IMAGES]
+train_masks = sorted(glob(os.path.join(DATA_DIR, "Category_ids/*")))[:MAX_IMAGES]
+
+
+def read_image(image_path, mask=False):
+    image = tf.io.read_file(image_path)
+    if mask:
+        image = tf.image.decode_png(image, channels=1)
+        image.set_shape([None, None, 1])
+        image = tf.image.resize(images=image, size=[IMAGE_SIZE, IMAGE_SIZE])
+    else:
+        image = tf.image.decode_png(image, channels=3)
+        image.set_shape([None, None, 3])
+        image = tf.image.resize(images=image, size=[IMAGE_SIZE, IMAGE_SIZE])
+        image = image / 127.5 - 1
+    return image
+
+
+def load_data(image_list, mask_list):
+    image = read_image(image_list)
+    mask = read_image(mask_list, mask=True)
+    return image, mask
+
+
+def data_generator(image_list, mask_list):
+    dataset = tf.data.Dataset.from_tensor_slices((image_list, mask_list))
+    dataset = dataset.map(load_data, num_parallel_calls=tf.data.AUTOTUNE)
+    dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)
+    return dataset
+
+
+dataset = data_generator(train_images, train_masks)
+dataset
+
+"""
+## Building DeepLabV3+ Model
+"""
+
+"""
+DeepLabv3+ extends DeepLabv3 by employing a encoder-decoder structure. The encoder module
+encodes multi-scale contextual information by applying atrous convolution at multiple
+scales, while the simple yet effective decoder module refines the segmentation results
+along object boundaries.
+
+![](https://github.com/lattice-ai/DeepLabV3-Plus/raw/master/assets/deeplabv3_plus_diagram.png)
+![](https://github.com/lattice-ai/DeepLabV3-Plus/raw/master/assets/deeplabv3_plus_diagram.png)
+
+**Dilated Convolution:** With Dilated convolution, as we go deeper, we can keep the
+stride constant but with larger field-of-view without increasing the number of parameters
+or the amount of computation. And finally, we can have larger output feature map which is
+good for semantic segmentation.
+
+The reason for using **Atrous Spatial Pyramid Pooling** is that it is discovered as the
```

Comment:
Change "atrous" to "dilated" everywhere.

Reply:
Changed atrous to dilated everywhere except the references.